crates/workspace/src/workspace_settings.rs

```diff
@@ -38,11 +38,14 @@ pub enum AutosaveSetting {
     /// Disable autosave.
     Off,
     /// Save after inactivity period of `milliseconds`.
-    AfterDelay { milliseconds: u64 },
+    AfterDelay {
+        milliseconds: u64,
+        trigger_formatter: Option<bool>,
+    },
     /// Autosave when focus changes.
-    OnFocusChange,
+    OnFocusChange { trigger_formatter: Option<bool> },
     /// Autosave when the active window changes.
-    OnWindowChange,
+    OnWindowChange { trigger_formatter: Option<bool> },
```

Comment:
Quick question without looking further at other code: isn't this a breaking change? i.e. configurations that already exist will be invalid?

Reply:
yes, this is a breaking change I should've noticed. If we give `OnFocusChange` and `OnWindowChange` types that are either string name or struct it should handle backwards compatibility. `trigger_formatter` will default to false if string name is given